server/src/main/java/org/opensearch/cluster/coordination/LeaderChecker.java

```diff
@@ -155,6 +159,10 @@ public void onNodeDisconnected(DiscoveryNode node, Transport.Connection connecti
         });
     }
 
+    private void setLeaderCheckTimeout(TimeValue newValue) {
+        this.leaderCheckTimeout = newValue;
+    }
```

Comment:
Similar comment as above. Should we add some checks to validate the `newValue`? For example, enforcing a minimum/maximum timeout value that we want to accept